File: interview/src/test/java/com/github/fanshuzaizai/interview/classLoad/Cat.java
package com.github.fanshuzaizai.interview.classLoad;

public class Cat {

    static {
        System.out.println("static....block initiated");
    }

    private final int age;

    public Cat() {
        this.age = 44;
    }

    public Cat(int age) {
        this.age = age;
    }

}

File: interview/src/test/java/com/github/fanshuzaizai/interview/scala/Cat.scala
package com.github.fanshuzaizai.interview.scala

/**
  * @author JiangZY
  * @date Created in 2019/9/8
  */
class Cat(name: String) {

  def sayHello: Unit = {
    println(name)
  }

}


object Cat {

  val name = "Tom"

  def apple: Cat = {
    new Cat(name)
  }

}
